### PR TITLE
Integrate CUDNN v8 frontend API for fused convolution

### PR DESCRIPTION
**Pull request by kaixih**
This PR integrates the new cuDNN Frontend API (https://github.com/NVIDIA/cudnn-frontend) for the fused convolution backends (i.e. Conv2D + BiasAdd + Relu).

cc. @nluehr

### CODE REVIEW

**Comment by kaixih** [comment]
@timshen91 Can you help review this one as well since it is related to the previous cudnn v8 frontend api PR?

File: tensorflow/stream_executor/dnn.h [hunk 1]
```diff
@@ -1383,6 +1383,15 @@ class DnnSupport {
       const dnn::ConvolutionDescriptor& convolution_descriptor,
       std::vector<std::unique_ptr<dnn::ConvolveExecutionPlan>>* out_exec_plans);
 
+  virtual bool GetFusedConvolveExecutionPlans(
```
Comment (timshen91) [inline tensorflow/stream_executor/dnn.h:1386]: Can you move this to `cuda_dnn.h` as well, similar to `DoConvolveXXX`? Similarly, they can be accessed via `dynamic_cast`.

Comment (timshen91) [inline tensorflow/stream_executor/dnn.h:1386]: It should return a `Status` object.

File: tensorflow/stream_executor/stream_executor_pimpl.h [hunk 2]
```diff
@@ -374,6 +374,15 @@ class StreamExecutor {
       const dnn::ConvolutionDescriptor &convolution_descriptor,
       std::vector<std::unique_ptr<dnn::ConvolveExecutionPlan>> *out_exec_plans);
 
+  bool GetFusedConvolveExecutionPlans(
```
Comment (timshen91) [inline tensorflow/stream_executor/stream_executor_pimpl.h:377]: It should return a `Status` object.

**Review by timshen91 — APPROVED** [review]

**Pull request merged**